app/controllers/content_items_controller.rb

```diff
@@ -22,6 +23,40 @@ def show
     render_template
   end
 
+  def choose_sign_in
+    set_up_self_assessment_ab_test
+    content_item = Services.content_store.content_item(content_item_path)
+    @content_item = ContentItemPresenter.new(content_item, content_item_path)
+    @error = params[:error]
+    render template: 'content_items/signin/choose-sign-in'
+  end
+
+  def not_registered
+    set_up_self_assessment_ab_test
```

Comment:
you could extract

```
    set_up_self_assessment_ab_test
    content_item = Services.content_store.content_item(content_item_path)
    @content_item = ContentItemPresenter.new(content_item, content_item_path)
```

Into another method and call it on the 3 actions you have

Reply:
Good point. Thanks.

Reply:
Extracted in commit a1a003851b1c2e4768f5157c2f8d9f6adb041078